src/main/java/org/folio/search/service/locationunit/CampusService.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.search.service.locationunit;
+
+import static org.apache.commons.collections4.MapUtils.getString;
+import static org.folio.search.utils.SearchUtils.CAMPUS_RESOURCE;
+import static org.folio.search.utils.SearchUtils.ID_FIELD;
+
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.search.client.CampusesClient;
+import org.folio.search.configuration.properties.ReindexConfigurationProperties;
+import org.folio.search.domain.dto.FolioIndexOperationResponse;
+import org.folio.search.domain.dto.ResourceEvent;
+import org.folio.search.service.ResourceService;
+import org.springframework.stereotype.Service;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class CampusService {
```

Comment:
It makes sense to combine it with LocationService. As well as CampusesClient combine with LocationClient. The logic of reindexing these entities is the same, the difference is only in HTTP requests.